auth/env_keychain_test.go

```diff
@@ -60,25 +65,127 @@ func testEnvKeychain(t *testing.T, when spec.G, it spec.S) {
 				})
 
 				it("returns an error", func() {
-					_, err := auth.EnvKeychain("CNB_REGISTRY_AUTH")
+					_, err := auth.NewEnvKeychain("CNB_REGISTRY_AUTH")
 					h.AssertNotNil(t, err)
 				})
 			})
 		})
 
 		when("environment variable is not set", func() {
 			it("returns an empty keychain", func() {
-				keychain, err := auth.EnvKeychain("CNB_REGISTRY_AUTH")
+				keychain, err := auth.NewEnvKeychain("CNB_REGISTRY_AUTH")
 				h.AssertNil(t, err)
 
-				h.AssertEq(t, keychain, &auth.ResolvedKeychain{
-					Auths: map[string]string{},
+				h.AssertEq(t, keychain, &auth.EnvKeychain{
+					AuthHeaders: map[string]string{},
 				})
 			})
 		})
 	})
 
-	when("InMemoryKeychain", func() {
+	when("EnvKeychain", func() {
+		when("#Resolve", func() {
```

Comment:
`EnvKeychain` and `ResolvedKeychain` tested separately because they are now distinct.